Zombie/src/main/java/dk/gruppeseks/bodtrd/zombie/ZombieProcessor.java

```diff
@@ -71,52 +71,41 @@ else if (playerCenter != null && velocity.getMagnitude() < AGGRO_RANGE)
                     {
                         zombieLastPathUpdate.put(zombie.getID(), System.currentTimeMillis());
 
-                        Stack<Position> path = _ai.getPath(zombieCenter, playerCenter, world);
-                        path.pop(); // We dont need the first route.
-                        zombiePath.put(zombie.getID(), path);
-                        velocity = new Vector2(zombieCenter, path.peek());
+                        Path path = _ai.getPath(zombieCenter, playerCenter, world);
+
+                        zombie.add(path);
+                        velocity = new Vector2(zombieCenter, path.peekNextPosition());
                     }
                     catch (NoPathException ex)
                     {
-                        System.out.println("NO PATH");
                         velocity.setMagnitude(MOVEMENT_SPEED);
                     }
                 }
-                Stack<Position> path = zombiePath.get(zombie.getID());
+                Path path = zombie.get(Path.class);
                 if (zombieLastPosition.get(zombie.getID()) != null && path != null)
                 {
-                    Vector2 currentVector = new Vector2(zombieCenter, path.peek());
+                    Vector2 currentVector = new Vector2(zombieCenter, path.peekNextPosition());
                     if (currentVector.getMagnitude() < 10) // Currently the fact that their position isn't at center is a problem, because sometimes it goes under 851-849.99 for example and it thinks its on the other grid. etc etc
                     {
-                        if (path.size() > 1)
+                        if (path.hasNext())
                         {
-                            path.pop();
+                            path.nextPosition();
                         }
 
                     }
-
-                    velocity = new Vector2(zombieCenter, path.peek());
-                    velocity.setMagnitude(MOVEMENT_SPEED);
-                    if (zombieId == zombie.getID())
+                    if (path.hasNext())
```

Comment:
is this supposed to be here? logic wise nothing alike was in previous commit. 
it looks like a flaw with the Path implementation